ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/state/beaconstate/versions/altair/ValidatorFlag.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.spec.datastructures.state.beaconstate.versions.altair;
+
+public class ValidatorFlag {
+  public static final int TIMELY_HEAD_FLAG = 1;
+  public static final int TIMELY_SOURCE_FLAG = 2;
+  public static final int TIMELY_TARGET_FLAG = 4;
+
+  public static boolean isTimelyTarget(int value) {
+    return areFlagsSet(value, TIMELY_TARGET_FLAG);
+  }
+
+  public static boolean isAnyFlagSet(int value) {
+    return areFlagsSet(value, TIMELY_HEAD_FLAG, TIMELY_SOURCE_FLAG, TIMELY_TARGET_FLAG);
+  }
+
+  private static boolean areFlagsSet(final int value, final int... flags) {
+    return (value & combineFlags(flags)) != 0;
+  }
```

Comment:
I feel like we should do this the other way around.  So this becomes `isFlagSet` so that `isTimelyTarget` has a cheap single bitwise or and then `isAnyFlagSet` would do the `combineFlags` before passing it in (or probably could just check if `value != 0`.

It's probably over-optimising but if we wind up checking single flags for all validators the extra CPU operations to calculate `combineFlags` starts to add up and likely gets inlined so never appears in a profiler.